src/lib/containers/Uploader.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from 'react'
+import { FileEntity } from '../utils/jsonResponses/FileEntity'
+import { uploadFile, createEntity, updateEntity, lookupChildEntity, getEntity } from '../utils/SynapseClient'
+import { FileUploadComplete } from '../utils/jsonResponses/FileUploadComplete'
+import { EntityId } from '../utils/jsonResponses/EntityId'
+
+type UploaderState = {
+  token?: string,
+  error?: any,
+  totalFilesToUploadCount: number,
+  filesUploadedCount: number,
+  isUploading?: boolean,
+  successfullyUploaded: boolean
+}
+
+export type UploaderProps = {
+  token?: string,
+  parentContainerId: string
+}
+
+export default class Uploader extends React.Component<UploaderProps, UploaderState> {
+  private readonly inputOpenFileRef: React.RefObject<HTMLInputElement>
+
+  constructor(props: UploaderProps) {
+    super(props)
+    this.state = {
+      token: '',
+      isUploading: false,
+      filesUploadedCount: 0,
+      totalFilesToUploadCount: 0,
+      successfullyUploaded: false
+    }
+    this.inputOpenFileRef = React.createRef()
+  }
+
+  showOpenFileDlg = () => {
+    if (this.inputOpenFileRef && this.inputOpenFileRef.current) {
+      this.inputOpenFileRef.current.click()
+    }
+  }
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.files) {
+      this.handleFilesChanged(event.target.files)
+    }
+  }
+
+  finishedProcessingOneFile = () => {
+    const incrementedFilesUploadedCount = this.state.filesUploadedCount + 1
+    const isStillUploading = incrementedFilesUploadedCount < this.state.totalFilesToUploadCount
+    this.setState(
+      {
+        isUploading: isStillUploading,
+        filesUploadedCount: incrementedFilesUploadedCount,
+        successfullyUploaded: !isStillUploading
+      })
+  }
+
+  handleUploadError = (error: any) => {
+    this.finishedProcessingOneFile()
+    this.setState({ error: { error } })
```

Comment:
I believe this should be this.setState({error}) ? otherwise the state contains a key 'error' who's value is a dictionary with a key named 'error' and the actual error value